tests/schema/schema.test.mjs

```diff
@@ -26,7 +26,64 @@ const parseYamlFromFile = (filePath) => {
 
 setMetaSchemaOutputFormat(BASIC);
 
-const validateOpenApi = await validate("./src/schemas/validation/schema.yaml");
+addKeyword({
+  id: "https://spec.openapis.org/oas/schema/vocab/keyword/discriminator",
+  interpret: (discriminator, instance, context) => {
+    return true;
+  },
+  /* discriminator is not exactly an annotation, but it's not allowed
+   * to change the validation outcome (hence returing true from interopret())
+   * and for our purposes of testing, this is sufficient.
+   */
+  annotation: (discriminator) => {
+    return discriminator;
+  },
+});
+
+addKeyword({
+  id: "https://spec.openapis.org/oas/schema/vocab/keyword/example",
+  interpret: (example, instance, context) => {
+    return true;
+  },
+  annotation: (example) => {
+    return example;
+  },
+});
+
+addKeyword({
+  id: "https://spec.openapis.org/oas/schema/vocab/keyword/externalDocs",
+  interpret: (externalDocs, instance, context) => {
+    return true;
+  },
+  annotation: (externalDocs) => {
+    return externalDocs;
+  },
+});
+
+addKeyword({
+  id: "https://spec.openapis.org/oas/schema/vocab/keyword/xml",
+  interpret: (xml, instance, context) => {
+    return true;
+  },
+  annotation: (xml) => {
+    return xml;
+  },
+});
+
+defineVocabulary(
+  "https://spec.openapis.org/oas/3.1/vocab/base",
+  {
+    "discriminator": "https://spec.openapis.org/oas/schema/vocab/keyword/discriminator",
+    "example": "https://spec.openapis.org/oas/schema/vocab/keyword/example",
+    "externalDocs": "https://spec.openapis.org/oas/schema/vocab/keyword/externalDocs",
+    "xml": "https://spec.openapis.org/oas/schema/vocab/keyword/xml",
+  },
+);
```

Comment:
You shouldn't need to define the vocabulary or keywords. They're already loaded when you import from `@hyperjump/json-schema/openapi-3-1`.

Reply:
> the problem is that we're using a schema with a different `$id` from the published 3.1 schema

Right. Registering the meta-schemas immediately after this block addresses that concern. It's just defining the vocabulary (`defineVocabulary`) and the keywords (`addKeyword`) that's unnecessary.

Reply:
Oh, yeah. I never added support for 3.2. I'll work on it.